esi_ui/api/esi_api.py

```diff
@@ -2,81 +2,122 @@
+        :type connection: :class:`~esi.connection.ESIConnection`
+        :returns: A list of dictionaries containing baremetal and lease information
+        """
+
+        esi_nodes = []
+        baremetal_nodes = []
+
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            f1 = executor.submit(connection.lease.nodes)
+            f2 = executor.submit(connection.baremetal.nodes, details=True)
+            esi_nodes = list(f1.result())
+            baremetal_nodes = list(f2.result())
+
+        esi_nodes.sort(key=lambda node: node.name)
+        baremetal_nodes.sort(key=lambda node: node.name)
+
+        return [
+            {
+                'uuid': bm.id,
+                'name': bm.name,
+                'maintenance': bm.is_maintenance,
+                'provision_state': bm.provision_state,
+                'target_provision_state': bm.target_provision_state,
+                'power_state': bm.power_state,
+                'target_power_state': bm.target_power_state,
+                'properties': bm.properties,
+                'lease_uuid': e.lease_uuid,
+                'owner': e.owner,
+                'lessee': e.lessee,
+                'resource_class': e.resource_class,
+                'future_offers': e.future_offers,
+                'future_leases': e.future_leases,
+            }
+            for e, bm in zip(esi_nodes, baremetal_nodes)
+        ]
 
-    esi_nodes = []
-    ironic_nodes = []
     with concurrent.futures.ThreadPoolExecutor() as executor:
-        esi_future = executor.submit(esiclient(token).list_nodes)
-        ironic_future = executor.submit(ironicclient(token).node.list, detail=True)
-        esi_nodes = esi_future.result()
-        ironic_nodes = ironic_future.result()
-
-    esi_nodes.sort(key=lambda node: node.name)
-    ironic_nodes.sort(key=lambda node: node.name)
-
-    return [
-        {
-            "name": ironic_node.name,
-            "uuid": ironic_node.uuid,
-            "provision_state": ironic_node.provision_state,
-            "target_provision_state": ironic_node.target_provision_state,
-            "power_state": ironic_node.power_state,
-            "target_power_state": ironic_node.target_power_state,
-            "maintenance": ironic_node.maintenance,
-            "lease_uuid": esi_node.lease_uuid,
-            "owner": esi_node.owner,
-            "lessee": esi_node.lessee,
-            "resource_class": esi_node.resource_class,
-        }
-        for esi_node, ironic_node in zip(esi_nodes, ironic_nodes)
-    ]
-
-
-def set_power_state(request, node_id, state):
+        f1 = executor.submit(nodes.network_list, connection)
+        f2 = executor.submit(node_list_full_info, connection)
+        node_networks = f1.result()
+        node_infos = f2.result()
+
+    for node in node_infos:
+        network_info = next((n['network_info'] for n in node_networks
+                             if n['node'].name == node['name']), None)
```

Comment:
Ah, this needs to be matched by name since the node_network function doesn't include node UUID information! This is fine for now, but I'd put in another comment that this will be updated to match on UUID after the matching esisdk function is updated.

Reply:
Acutally, since the node_network function returns the baremetal node object, it does include the node UUID. So this can be fixed.

Reply:
Ah, great!